packages/TimePicker/src/TimePicker.js

```diff
@@ -137,94 +186,72 @@ export default class TimeInput extends React.Component {
+      finish();
     }
 
     if (event.key === "Escape") {
-      this.finish();
+      finish();
     }
   };
 
-  handleError(time) {
-    this.setState({
-      ...time,
-      error: this.props.labelError,
-      isVisible: false,
-    });
-  }
-
-  finish() {
-    this.setState(state => ({
-      isVisible: false,
-      value: state.timeStr,
-    }));
-  }
-
-  render() {
-    const {
-      isDisabled,
-      ariaLabel,
-      labelCustom,
-      labelHours,
-      labelMinutes,
-      labelPeriod,
-      labelAM,
-      labelPM,
-      ...moreProps
-    } = this.props;
-
-    return (
-      <sc.CSSHolder>
-        <scPicker.CSSHolder>
-          <div className="timeinput" onFocus={this.handleFocus} onBlur={this.handleBlur}>
-            {/* eslint-disable-next-line */}
-            <div tabIndex={this.state.isTabIndexActive ? 0 : -1}>
-              {/* setting hasClearButton to false confuses, look like a close button for the timeinput hasClearButton */}
-              <div ref={this.$refInputContainer}>
-                <Input
-                  ariaLabel={ariaLabel}
-                  hasClearButton={false}
-                  isDisabled={isDisabled}
-                  onInput={this.handleChange}
-                  onFocus={this.handleFocus}
-                  onKeyUp={this.handleKeyUp}
-                  value={this.state.value}
-                  data-qa-id="time-input__starting-at"
-                  {...moreProps}
-                />
-              </div>
-              {/** THIS ERROR SHOULD BE REPLACE WITH A PROPER FORM-ELEMENT */}
-              <span>{this.state.error}</span>
-              <Picker
-                hh={this.state.hh}
-                isVisible={this.state.isVisible}
-                labelCustom={labelCustom}
-                labelHours={labelHours}
-                labelMinutes={labelMinutes}
-                labelPeriod={labelPeriod}
-                labelAM={labelAM}
-                labelPM={labelPM}
-                mm={this.state.mm}
-                onClick={this.handleClick}
-                period={this.state.period}
+  return (
+    <L10n>
+      <sc.TimePicker onFocus={handleFocus} onBlur={handleBlur}>
+        <Popover style={{ width: "100%" }} isOpen={isOpen} edge="left" offset={0} align="bottom">
+          {/* eslint-disable-next-line */}
+          <div tabIndex={isTabIndexActive ? 0 : -1}>
+            {/* setting hasClearButton to false confuses, look like a close button for the timeinput hasClearButton */}
+            <Popover.Trigger style={{ width: "100%" }}>
+              <Input
+                ariaLabel={a11yText}
+                hasClearButton={false}
+                icon={<SvgClockTime />}
+                isDisabled={isDisabled}
+                onChange={handleChange}
+                onKeyUp={handleKeyUp}
+                onFocus={handleFocus}
+                value={value}
+                data-qa-id="time-input__starting-at"
+                data-pka-anchor="timePicker-Input"
```

Comment:
I think we can remove the `data-qa-id` cc @nahumzs

Reply:
yeap we can remove it, just let's check we are no using it on any testing.